lib/util/linkComponents.js

```diff
@@ -12,7 +12,9 @@ const DEFAULT_LINK_ATTRIBUTE = 'href';
 function getLinkComponents(context) {
   const settings = context.settings || {};
   const linkComponents = /** @type {typeof DEFAULT_LINK_COMPONENTS} */ (
-    DEFAULT_LINK_COMPONENTS.concat(settings.linkComponents || [])
+    settings.linkComponents
+      ? DEFAULT_LINK_COMPONENTS.concat(settings.linkComponents)
+      : DEFAULT_LINK_COMPONENTS
```

Comment:
repeating `DEFAULT_LINK_COMPONENTS` seems unfortunate; is creating an empty array really that costly?

Reply:
Creating a copy of an array should be basically free.